framework/py/flwr/superlink/auth_plugin/noop_auth_plugin.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2025 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Concrete NoOp implementation for Servicer-side account authentication and
+authorization plugins."""
+
+
+from collections.abc import Sequence
+from pathlib import Path
+from typing import Optional, Union
+
+from flwr.common.constant import NOOP_ACCOUNT_NAME, NOOP_FLWR_AID, AuthType
+from flwr.common.typing import AccountInfo, UserAuthCredentials, UserAuthLoginDetails
+
+from .auth_plugin import ControlAuthPlugin, ControlAuthzPlugin
+
+NOOP_ACCOUNT_INFO = AccountInfo(
+    flwr_aid=NOOP_FLWR_AID,
+    account_name=NOOP_ACCOUNT_NAME,
+)
+
+
+class NoOpControlAuthPlugin(ControlAuthPlugin):
+    """No-operation implementation of ControlAuthPlugin."""
+
+    def __init__(
+        self,
+        user_auth_config_path: Path,
+        verify_tls_cert: bool,
+    ):
+        pass
+
+    def get_login_details(self) -> Optional[UserAuthLoginDetails]:
+        """Get the login details."""
+        return UserAuthLoginDetails(
+            auth_type=AuthType.NOOP,  # No operation auth type
+            device_code="",
+            verification_uri_complete="",
+            expires_in=0,
+            interval=0,
+        )
```

Comment:
Should we actually return something here? Or raise a specific exception that causes the interceptor to return a specific error code?